### PR TITLE
Add handwired/justageng/Office60 keyboard

### PR DESCRIPTION
**Pull request by JiaGengChang**
## Description

Adding stm32 blue pill handwired keyboard

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/justageng/office60/readme.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
+# Office60
+
+![Office60](imgur.com image replace me!)
```
Comment (keyboard-magpie) [inline keyboards/handwired/justageng/office60/readme.md:3]: Please tidy or replace this appropriately.

File: keyboards/handwired/justageng/office60/readme.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+Failed to retrieve language identifiers
+Opening DFU capable USB device...
+Device ID 1eaf:0003
+Device DFU version 0110
+Claiming USB DFU Interface...
+Setting Alternate Interface #2 ...
+Cannot set alternate interface: LIBUSB_ERROR_IO
+```
+
+You probably got a lemon from the manufacturer, and I do not know how to fix this apart from buying a new bluepill and praying that it does not come with the incompatible version. This sour lemon bluepill also gives a strange output for dfu-util list when the BOOT1 jumper is disconnected (if BOOT1 jumper is connected, the --list output is identical to the compatible bluepill - qmk firmware will not be flashed successfully even if command prompt says its successful):
```
Comment (keyboard-magpie) [inline keyboards/handwired/justageng/office60/readme.md:107]: Colloquialisms like "a lemon" may not be understood or translated very well, please change to "a defective unit" or similar.

File: keyboards/handwired/justageng/office60/readme.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
+
+## Bootloader
+
+Enter the bootloader the following 2 ways:
+* **Bootmagic**: Hold down the escape key, or the key at the top left corner
+* **Physical reset button**: Briefly press the reset button on the bluepill 
+
+
+Windows 10 instructions for flashing firmware:
```
Comment (drashna) [inline keyboards/handwired/justageng/office60/readme.md:28]: Most of this looks to be just stm32duino flashing instructions, which are posted elsewhere.  

For instance: 

https://github.com/rogerclarkmelbourne/Arduino_STM32/wiki/Flashing-Bootloader-for-BluePill-Boards

File: keyboards/handwired/justageng/office60/rules.mk [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output
+
+DEFAULT_FOLDER = handwired/justageng/office60
+
+# Enter lower-power sleep mode when on the ChibiOS idle thread
+OPT_DEFS += -DCORTEX_ENABLE_WFI_IDLE=TRUE
+
+# enable link time optimization
+EXTRAFLAGS += -flto
```
Comment (drashna) [inline keyboards/handwired/justageng/office60/rules.mk:29]:
```suggestion
LTO_ENABLE = yes
```

File: keyboards/handwired/justageng/office60/readme.md (hunk 3 above)
Comment (JiaGengChang) [reply keyboards/handwired/justageng/office60/readme.md]: Do you prefer if I cut down on the length of the readme, and point users to other sites for instructions?

Comment (JiaGengChang) [reply keyboards/handwired/justageng/office60/readme.md]: I have added the link you provided to the readme for additional reference.

**Review by keyboard-magpie — APPROVED** [review]

File: keyboards/handwired/justageng/office60/config.h [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+#pragma once
+
+#include "config_common.h"
+
+#define VENDOR_ID       0x0FF1
+#define PRODUCT_ID      0xCE60
+#define DEVICE_VER      0x0001
+
+#define MANUFACTURER    Jia Geng
+#define PRODUCT         Office60
```
Comment (drashna) [inline keyboards/handwired/justageng/office60/config.h:26]:
```suggestion
```

File: keyboards/handwired/justageng/office60/info.json [hunk 6]
```diff
@@ -0,0 +1,78 @@
+{
+    "keyboard_name" : "office60",
+    "bootloader": "stm32duino",
+    "processor": "STM32F103",
+    "url": "https://github.com/JiaGengChang/qmk_firmware",
+    "maintainer": "JiaGengChang",
```
Comment (drashna) [inline keyboards/handwired/justageng/office60/info.json:6]: 0x0FF1 is in use:
https://www.the-sz.com/products/usbid/index.php?v=0x0FF1&p=&n=

In general, the further past 0x4000 the VID is, the better.  I've set it to FEED, for the meanwhile. 

```suggestion
    "maintainer": "JiaGengChang",
    "usb": {
        "vid": "0xFEED",
        "pid": "0xCE60",
        "device_version": "0.0.1"
    },
```

File: keyboards/handwired/justageng/office60/readme.md [hunk 7]
```diff
@@ -0,0 +1,126 @@
+# Office60
+
+*A short description of the keyboard/project*
```
Comment (drashna) [inline keyboards/handwired/justageng/office60/readme.md:3]: This needs to be fixed

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/handwired/justageng/office60/readme.md (hunk 7 above)
Comment (JiaGengChang) [reply keyboards/handwired/justageng/office60/readme.md]: I've replaced the line with a description of the keyboard.

File: keyboards/handwired/justageng/office60/config.h [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+#define MATRIX_ROW_PINS \
+    { A0, A1, A2, A3, A4 }
+#define UNUSED_PINS \
+    { A5, A6, A7, A13, A14, B0, B1, B2, B10, B11, B15, C13, C14, C15 }
+
+
+// disable more qmk features
+#define NO_ACTION_LAYER
+#define NO_ACTION_TAPPING
+#define NO_ACTION_ONESHOT
```
Comment (drashna) [inline keyboards/handwired/justageng/office60/config.h:37]: Is there a reason for disabling basically everything here?

File: keyboards/handwired/justageng/office60/config.h [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+#define IGNORE_MOD_TAP_INTERRUPT
```
Comment (drashna) [inline keyboards/handwired/justageng/office60/config.h:38]: This should be moved to the keymap level

**Review by drashna — CHANGES_REQUESTED** [review]

**Pull request closed without merging**